Tests/SwiftDocCTests/Semantics/ArticleTests.swift

```diff
@@ -208,4 +208,45 @@ class ArticleTests: XCTestCase {
         
         XCTAssertEqual(article?.options[.local]?.automaticSeeAlsoEnabled, false)
     }
+    
+    func testDisplayNameDirectiveIsRemoved() throws {
+        let source = """
+        # Root
+        
+        @Metadata {
+          @TechnologyRoot
+          @PageColor(purple)
+          @DisplayName("Example")
+        }
+        
+        Adding @DisplayName to an article will result in a warning.
+        """
+        let document = Document(parsing: source, options: [.parseBlockDirectives])
+        let (bundle, context) = try testBundleAndContext()
+        var problems = [Problem]()
+        let article = Article(from: document, source: nil, for: bundle, in: context, problems: &problems)
+        
+        XCTAssertEqual(problems.map(\.diagnostic.summary), [
+            "A 'DisplayName' directive is only supported in documentation extension files. To customize the display name of an article, change the content of the level-1 heading."
+        ])
+        
+        let semantic = try XCTUnwrap(article)
+        XCTAssertNotNil(semantic.metadata, "Article has a @Metadata directive")
```

Comment:
This may be more of an XCTest question than anything, but what does the second parameter represent? The docs describe it as [a description of a failure](https://developer.apple.com/documentation/xctest/1500453-xctassertnotnil), but then I would expect the description here to be that the Article object does _not_ have metadata (but should). Am I understanding this incorrectly?

Reply:
That's a good question. I mainly think of it as extra information for the test failure error message, especially if the description contains dynamic values from the test execution. Thinking about it some more I probably use it a bit inconsistently. 

In most cases I find that the test code reads better if the description is a statement about what should be true rather than a statement about what wasn't true. For example
```swift 
XCTAssertNotNil(semantic.metadata, "Article should have a Metadata configuration")
```
instead of 
```swift 
XCTAssertNotNil(semantic.metadata, "Missing expected Metadata configuration")
```
but at the same time I prefer to use statements about what wasn't true when asserting that an operation didn't report any problems. 
```swift 
XCTAssert(context.problems.isEmpty, Unexpected problems "\(context.problems.map(\.diagnostic.summary))")
```

In this description I skipped over _what_ should be true and instead described _why_ the assertion should be true but upon reading it again that becomes confusing because it can also be interpreted as a statement about what should be true. 

-----

Now that I think about it, my inconsistent use of how I use the description probably comes from using it for two purposes:
- as a failure description
- as a comment about what the assertion is verifying

If I should have been really strict about it I would have probably written the assertion like:
```swift
// The article should have a metadata container because the markup contains a @Metadata directive
XCTAssertNotNil(semantic.metadata, "Missing expected metadata configuration")
```

but in that case the failure description is just a spelled out version of the information in `XCTAssertNotNil(semantic.metadata)` so I'd be inclined to remove it but that leaves me with a comment and no failure description. 
```swift
// The article should have a metadata container because the markup contains a @Metadata directive
XCTAssertNotNil(semantic.metadata)
```

In this case, having this information in a comment or a failure description doesn't make much difference since someone would likely need to open the code to investigate the issue if this test ever failed. 

But since there is a description argument that shows up in the test failure error message I like to include this information there in case the extra context helps someone understand a test failure better/faster.

In that case the last question is if I should rephrase the information to describe what the failure or if the information should be a statement about what's expected. In both cases the _why_ is likely more important than the _what_ and I usually find that "expected BEHAVIOR because REASON" sentences read better than "unexpected BEHAVIOR because REASON" sentences. For example:

```swift
XCTAssertNotNil(semantic.metadata, "Article should have a metadata container because the markup has a @Metadata directive")
```

```swift
XCTAssertNotNil(semantic.metadata, "Missing metadata container that's expected because the markup has a @Metadata directive")
```

----


I updated these descriptions to be statements about what should be true and why in https://github.com/apple/swift-docc/pull/783/commits/d93c93a0d266a76b72ca612ffacff1254a9ed21b which still isn't a good _failure_ description but, for the above reasons, I find that phrasing to be more readable.

Reply:
Ah yup, I was thinking of the _what_ not the _why_. That makes sense, thanks for the detailed explanation!